deadlock_demolition.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: doc
+title: "Deadlock Demolition"
+learning_objectives:
+  - Synchronization Primitives
+  - Deadlock Detection using a Resource Allocation Graph
```

Comment:
Split this up to deadlock and resource allocation graphs as two different points?